### PR TITLE
Edit pass for grpc browser.md & move snippets to code files

### PR DESCRIPTION
**Pull request by wadepickett**
[Internal Review](https://review.docs.microsoft.com/en-us/aspnet/core/grpc/browser?view=aspnetcore-3.0&branch=pr-en-us-16676)

Edit pass for pr #16613 for grpc topic browser.md,
Making localization friendly: Moving snippets out to code files while moving code comments over into instruction.

### CODE REVIEW

File: aspnetcore/grpc/browser.md [hunk 1]
```diff
@@ -88,7 +88,7 @@ var response = await client.SayHelloAsync(new GreeterRequest { Name = ".NET" });
 
 The `GrpcWebHandler` has some configuration options when it is created:
```
Comment (Rick-Anderson) [inline aspnetcore/grpc/browser.md]:
```suggestion
`GrpcWebHandler` has the following configuration options when created:
```

File: aspnetcore/grpc/browser.md [hunk 2]
```diff
@@ -20,37 +20,21 @@ It is not possible to call a HTTP/2 gRPC service from a browser-based app. [gRPC
 
 ## Configure gRPC-Web in ASP.NET Core
 
-gRPC services hosted in ASP.NET Core can be configured to support gRPC-Web alongside HTTP/2 gRPC. gRPC-Web does not require any changes to services, the only modification is startup configuration.
+gRPC services hosted in ASP.NET Core can be configured to support gRPC-Web alongside HTTP/2 gRPC. gRPC-Web does not require any changes to services. The only modification is startup configuration.
 
 To enable gRPC-Web with an ASP.NET Core gRPC service:
 
 1. Add a reference to the [Grpc.AspNetCore.Web](https://www.nuget.org/packages/Grpc.AspNetCore.Web) package.
```
Comment (Rick-Anderson) [inline aspnetcore/grpc/browser.md]: Use *, not 1, 2
You don't need to do those steps in that order. Even if you do, we prefer *.  When you're forced to use numbers, use 1. each time. 
```
1.
1.
1.
```
The build system will number them. Updates don't need to be edited.

File: aspnetcore/grpc/browser.md [hunk 3]
```diff
@@ -20,37 +20,21 @@ It is not possible to call a HTTP/2 gRPC service from a browser-based app. [gRPC
...
 2. Configure the app to use gRPC-Web by adding `AddGrpcWeb(...)` and `UseGrpcWeb()` to *Startup.cs*:
 
-```csharp
-public void ConfigureServices(IServiceCollection services)
-{
-    services.AddGrpc();
-}
+[!code-csharp[](~/grpc//browser/sample/Startup.cs?name=snippet_1&highlight=4,11,15)]
```
Comment (Rick-Anderson) [inline aspnetcore/grpc/browser.md]: Check the build. This is not showing up. First thing you do is check for build warnings.

File: aspnetcore/grpc/browser.md [hunk 4]
```diff
@@ -20,37 +20,21 @@ It is not possible to call a HTTP/2 gRPC service from a browser-based app. [gRPC
 
 ## Configure gRPC-Web in ASP.NET Core
 
-gRPC services hosted in ASP.NET Core can be configured to support gRPC-Web alongside HTTP/2 gRPC. gRPC-Web does not require any changes to services, the only modification is startup configuration.
+gRPC services hosted in ASP.NET Core can be configured to support gRPC-Web alongside HTTP/2 gRPC. gRPC-Web does not require any changes to services. The only modification is startup configuration.
 
 To enable gRPC-Web with an ASP.NET Core gRPC service:
 
 1. Add a reference to the [Grpc.AspNetCore.Web](https://www.nuget.org/packages/Grpc.AspNetCore.Web) package.
 2. Configure the app to use gRPC-Web by adding `AddGrpcWeb(...)` and `UseGrpcWeb()` to *Startup.cs*:
```
Comment (Rick-Anderson) [inline aspnetcore/grpc/browser.md]:
```suggestion
* Configure the app to use gRPC-Web by adding `AddGrpcWeb` and `UseGrpcWeb` to *Startup.cs*:
```

File: aspnetcore/grpc/browser.md (hunk 3 above)
Comment (Rick-Anderson) [reply aspnetcore/grpc/browser.md]: You've got `//` in the path.

File: aspnetcore/grpc/browser.md (hunk 4 above)
Comment (Rick-Anderson) [reply aspnetcore/grpc/browser.md]: That's only a suggestion.  Check the code to see what's actually called. I don't think `UseGrpcWeb` is called.

You need to make two sample's, the one he has and an alternate sample.

File: aspnetcore/grpc/browser.md [hunk 5]
```diff
@@ -20,37 +20,21 @@ It is not possible to call a HTTP/2 gRPC service from a browser-based app. [gRPC
...
 
-    // Add gRPC-Web middleware after routing and before endpoints
-    app.UseGrpcWeb();
+* Adds the gRPC-Web middleware, `UseGrpcWeb`, after routing and before endpoints.
+* Specifies the `endpoints.MapGrpcService<GreeterService>()` method supports gRPC-Web with `EnableGrpcWeb()`. Alternatively, configure all services to support gRPC-Web by adding `services.AddGrpcWeb(o => o.GrpcWebEnabled = true);` to ConfigureServices.
 
-    app.UseEndpoints(endpoints =>
-    {
-        // Specify that this method should support gRPC-Web with `EnableGrpcWeb()`.
-        // Alternatively, you can configure all services to support gRPC-Web by adding
```
Comment (Rick-Anderson) [inline aspnetcore/grpc/browser.md]: Make a snippet with this approach

File: aspnetcore/grpc/browser.md [hunk 6]
```diff
@@ -66,33 +50,27 @@ There is a JavaScript gRPC-Web client. For instructions on how to use gRPC-Web f
...
+[!code-csharp[](~/grpc/browser/sample/Handler.cs?name=snippet_1)]
+
+The preceding code:
 
-// Create a client and make calls using the channel
-var client = Greeter.GreeterClient(channel);
-var response = await client.SayHelloAsync(new GreeterRequest { Name = ".NET" });
-```
+* Configures a channel to use gRPC-Web.
+* Creates a client and makes a call using the channel.
```
Comment (Rick-Anderson) [inline aspnetcore/grpc/browser.md]: The server must have been configured to support gRPC-Web.

change that to
The server must be configured to support gRPC-Web.

File: aspnetcore/grpc/browser.md [hunk 7]
```diff
@@ -20,37 +20,21 @@ It is not possible to call a HTTP/2 gRPC service from a browser-based app. [gRPC
...
 
-public void Configure(IApplicationBuilder app)
-{
-    app.UseRouting();
+The preceding code:
 
-    // Add gRPC-Web middleware after routing and before endpoints
-    app.UseGrpcWeb();
+* Adds the gRPC-Web middleware, `UseGrpcWeb`, after routing and before endpoints.
+* Specifies the `endpoints.MapGrpcService<GreeterService>()` method supports gRPC-Web with `EnableGrpcWeb()`. Alternatively, configure all services to support gRPC-Web by adding `services.AddGrpcWeb(o => o.GrpcWebEnabled = true);` to ConfigureServices.
```
Comment (Rick-Anderson) [inline aspnetcore/grpc/browser.md]: `endpoints.MapGrpcService<GreeterService>()` is OK and an exception to the rule. NEVER keep `()` in functions. It should be
`EnableGrpcWeb` not
`EnableGrpcWeb()`

Comment (Rick-Anderson) [inline aspnetcore/grpc/browser.md]: Alternatively
Provide an alternative snippet.

File: aspnetcore/grpc/browser.md [hunk 8]
```diff
@@ -66,33 +50,27 @@ There is a JavaScript gRPC-Web client. For instructions on how to use gRPC-Web f
...
 2. Configure the channel to use the `GrpcWebHandler`:
 
-```csharp
-// Configure a channel to use gRPC-Web
-var handler = new GrpcWebHandler(GrpcWebMode.GrpcWebText, new HttpClientHandler());
-var channel = GrpcChannel.ForAddress("https://localhost:5001", new GrpcChannelOptions
-    {
-        HttpClient = new HttpClient(handler)
-    });
+[!code-csharp[](~/grpc/browser/sample/Handler.cs?name=snippet_1)]
```
Comment (Rick-Anderson) [inline aspnetcore/grpc/browser.md]: snippet not showing up.

File: aspnetcore/grpc/browser.md [hunk 9]
```diff
@@ -66,33 +50,27 @@ There is a JavaScript gRPC-Web client. For instructions on how to use gRPC-Web f
...
-var response = await client.SayHelloAsync(new GreeterRequest { Name = ".NET" });
-```
+* Configures a channel to use gRPC-Web.
+* Creates a client and makes a call using the channel.
 
-The `GrpcWebHandler` has some configuration options when it is created:
+The `GrpcWebHandler` has the following configuration options when created:
 
-* **InnerHandler** - The underlying <xref:System.Net.Http.HttpMessageHandler> that will make the HTTP call, e.g. `HttpClientHandler`.
+* **InnerHandler** - The underlying <xref:System.Net.Http.HttpMessageHandler> that will make the HTTP call, for example, `HttpClientHandler`.
```
Comment (Rick-Anderson) [inline aspnetcore/grpc/browser.md]: Use `:`, not `-`
* **InnerHandler** : The ...

Comment (Rick-Anderson) [reply aspnetcore/grpc/browser.md]: Per style guide, avoid future tense
that will make the HTTP call, -> that **makes** the HTTP call,

File: aspnetcore/grpc/browser.md [hunk 10]
```diff
@@ -66,33 +50,27 @@ There is a JavaScript gRPC-Web client. For instructions on how to use gRPC-Web f
...
 * **Mode** - `GrpcWebMode` enum. `GrpcWebMode.GrpcWebText` configures content to be base64 encoded, which is required to support server streaming calls.
```
Comment (Rick-Anderson) [inline aspnetcore/grpc/browser.md]:
```suggestion
* **Mode** :  `GrpcWebMode` enum. `GrpcWebMode.GrpcWebText` configures content to be base64 encoded. Base64 encoding is required to support server streaming calls.
```

File: aspnetcore/grpc/browser/sample/Startup.cs [hunk 11]
```diff
@@ -0,0 +1,18 @@
+﻿#region snippet_1
```
Comment (Rick-Anderson) [inline aspnetcore/grpc/browser/sample/Startup.cs]: Create a full startup file you can compile.

File: aspnetcore/grpc/browser/sample/Handler.cs [hunk 12]
```diff
@@ -0,0 +1,10 @@
+#region snippet_1
```
Comment (Rick-Anderson) [inline aspnetcore/grpc/browser/sample/Handler.cs]: Create a real handler file

File: aspnetcore/grpc/browser.md (hunk 4 above)
Comment (wadepickett) [reply aspnetcore/grpc/browser.md]: Verified: UserGrpWeb() is called in the current full browser app sample.

Comment (Rick-Anderson) [reply aspnetcore/grpc/browser.md]: But it's not called in that snippet. You're doing two snippets - right?

Comment (wadepickett) [reply aspnetcore/grpc/browser.md]: In the original PR UseGrpWeb() is called:
 "// Add gRPC-Web middleware after routing and before endpoints
    app.UseGrpcWeb();"

In the full app sample it is also called in startup.cs
"app.UseGrpcWeb();"

Sidenote: I will be replacing the the small non-compilable snippet with the full startup.cs example in a moment.  Not ready for review yet.

File: aspnetcore/grpc/browser/sample/Handler.cs (hunk 12 above)
Comment (wadepickett) [reply aspnetcore/grpc/browser/sample/Handler.cs]: Noting what we decided to do with this one:  The app example available at the moment demonstrating how to properly set this up does not match close enough the snippet provided that I can create a real handler file I can trust.  For now, going with JamesNK's code snippit as-is.  However for the startup.cs example I created a compilable full file given that JamesNK's code snippet for that one close matched the guiding app example.

**Comment by wadepickett** [comment]
@Rick-Anderson, Thanks for the fantastic review help.  I resolved all items.  As we discussed earlier and as I noted inline above, while I was able to move the browser snippet into a startup.cs example that JamesNK had for guidance, I used the original snippet for client example moved to a code file.  The Blazor example to use for guidance varied from what JamesNK needs here a bit too much for me to use that.

File: aspnetcore/grpc/browser/sample/AllServicesSupportExample_Startup.cs [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+using Server.Services;
+
+namespace Server
+{
+    #region snippet_1
+
```
Comment (Rick-Anderson) [inline aspnetcore/grpc/browser/sample/AllServicesSupportExample_Startup.cs]: remove empty lines around snippets

**Comment by Rick-Anderson** [comment]
@JamesNK  once you approve, we'll S&M this into your PR, then wait for your instructions to make that live.

**Comment by Rick-Anderson** [comment]
@wadepickett  review [my commit](https://github.com/aspnet/AspNetCore.Docs/pull/16676/commits/dedbb6f9cb706aada9a4461fbb5c8e17c7bc92fe) for what we like. No long sentences.

File: aspnetcore/grpc/browser/sample/AllServicesSupportExample_Startup.cs (hunk 13 above)
Comment (wadepickett) [reply aspnetcore/grpc/browser/sample/AllServicesSupportExample_Startup.cs]: Removing empty lines on snippets, and will update the number highlighting in a moment to reflect the change.

**Pull request merged**